\documentclass[preprint,3p,times,twocolumn]{elsarticle}
\usepackage{graphicx}
\usepackage{color}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{multirow}
\usepackage{rotating,capt-of}
\usepackage[small]{caption}
\usepackage{booktabs}
\usepackage{float} % for placing figures where i want
\usepackage{afterpage}
\usepackage{epsfig, a4wide}

\newcommand{\myparagraph}[1]{
  \paragraph*{\normalfont\itshape #1}\hspace{5pt}}

% strange snos
\definecolor{purple}{RGB}{180,90,200}
\definecolor{dgreen}{RGB}{0,160,0}
\definecolor{turquoise}{RGB}{0,180,140}
\renewcommand\dblfloatpagefraction{0.03}
\renewcommand\topfraction{.95}
\renewcommand\bottomfraction{.95}
\renewcommand\textfraction{.05}
\renewcommand\floatpagefraction{.95}
\renewcommand\dbltopfraction{.95}
\renewcommand\dblfloatpagefraction{.95}
\newcommand{\TODO}[1] {\begingroup\color{red}#1\endgroup}
%\newcommand{\FINAL}[1]{\begingroup\color{dgreen}#1\endgroup}
\newcommand{\ACC}[1]{\emph{\textbf{#1}}}
\newcommand{\s}[1]{\begin{tiny}#1\end{tiny}}
\newcommand{\url}[1]{\texttt{\small #1}}
\newcommand{\maxentscan}{\texttt{MaxEntScan}}
\newcommand{\NEW}[1]{\begingroup\color{black}#1\endgroup}

%% abbreviations
\newcommand{\snos}{snoRNAs}
\newcommand{\sno}{snoRNA}
\newcommand{\cd}{box C/D snoRNA}
\newcommand{\haca}{box H/ACA snoRNA}
%% programs
\newcommand{\spps}{\texttt{SPPS}}
\newcommand{\tri}{\texttt{TRI\_tool}}
\newcommand{\lr}{\texttt{LR\_PPI}}

%% databases
\newcommand{\ncbi}{\texttt{NCBI}}




%FUNGI
\newcommand{\Hsa}{\emph{Homo sapiens}}
\newcommand{\hsa}{\emph{H.sapiens}}

%Trennung verhindern
\hyphenation{snoRNAs}
\hyphenation{snoRNA}
\hyphenation{microRNA}
\hyphenation{microRNAs}
\hyphenation{miRNA}
\hyphenation{miRNAs}
\hyphenation{lncRNA}
\hyphenation{lncRNAs}
\hyphenation{lincRNA}
\hyphenation{lincRNAs}
\hyphenation{di-nucleo-tide}


\journal{Preprint}

\DeclareCaptionLabelFormat{simplesupp}{#1~S#2} % new caption format with Sxx

\begin{document}

\begin{frontmatter}



\title{Predicting Protein-Protein-Interactions based on Autocorrelation and Random Forests}

\author[LEI]{Sebastian Canzler\corref{cor1}}
\ead{sebastian@bioinf.uni-leipzig.de}
\author[PHY]{Ren\'{e} Staritzbichler}
\ead{rene.staritzbichler@medizin.uni-leipzig.de}


\address[LEI]{Bioinformatics Group, Department of Computer Science,
  Leipzig University,
  H{\"a}rtelstra{\ss}e 16-18, D-04107 Leipzig, Germany
}
\address[PHY]{Institute of Medical Physics and Biophysics, University Leipzig,
    H{\"ä}rtelstraße 16-18, D-04107 Leipzig, Germany.}

%\cortext[jfa]{Joint first authors}
\cortext[cor1]{Corresponding author}

% --------------------------------------------------------------------------- %


\begin{abstract}
  
\end{abstract}

\begin{keyword}
  protein-protein-interaction, machine learning, random forest
\end{keyword}

\end{frontmatter}

% --------------------------------------------------------------------------- %

\section{Introduction}
The driving force of molecular networks is based in protein
interactions rather than single protein components accomplishing
biological functions \cite{Pawson:2004}. Therein, proteins of various
biological processes, such as cellular organization, regulation of
transcription and translation, or immune response need to interact and
work together to function appropriately. The experimental
identification of interacting proteins is highly expensive and time
consuming. A reliable \textit{in silico} prediction of protein-protein
interactions (PPI), however, will therefore shed some more
light in understanding biological and pharmacological responses and pathways. A
computational approach of identifying potential PPIs is mostely based
on an extensive set of known protein-protein interactions, information
about cellular localizations, amino acid sequences, or secondary
structures. Such methods may include phylogenetic trees
\cite{Pazos:2001}, phylogenetic profiles \cite{Barker:2005},
network-based methods \cite{Yook:2004, Clauset:2008}. In recent years,
the more sophisticated approach of combining distinct prediction
methods has been applied, e.g., \texttt{STRING} \cite{Szklarczyk:2011}
or \texttt{PIPS} \cite{McDowall:2009}. Nevertheless, different proteome-wide prediction methods have shown
that knowledge of the amino acid sequence alone may be sufficient to identify novel,
functional protein-protein interactions \cite{Martin:2005,
  Shen:2007}. Due to its prominent and major advantages like simplicity, rapidity,
and generality, this kind or prediction method became increasingly
widespread over the last years \cite{Ofran:2003, Betel:2007, Liu:2012,
  Perovic:2017, Pan:2010}.

In this paper, we present a sequence-based webtool for predicting PPIs
that outperforms currently available competitors. Mainly because of
fine-tuning of machine learning keystones, such as dataset generation,
selection of most informative amino acid scales, or the design of the
feature vector. These improvements in combination with a random forest
(RF) machine learning algorithm boost our predictive power to be a
cutting-edge, high-throughput, and extremley fast method. Therefore,
\TODO{OUR\_TOOL} may serve as a reliable tool to identify potential
interacting protein partner amongst the set of known proteins, and may
thus help to identify the yet unknown biological mechanism for several
existings proteins.

% --------------------------------------------------------------------------- %

\section{Materials and Methods}
\subsection{Feature vector calculation}

\subsection{Collecting datapoints}

% --------------------------------------------------------------------------- %

\section{Results}
%\input{results}
% --------------------------------------------------------------------------- %

\subsection{Comparison to other tools}
To verify our achievements, we compared our program against publicly available tools such as \spps\ \cite{Liu:2012}, \tri\ \cite{Perovic:2017}, and \lr\ \cite{Pan:2010}.

\section{Discussion}
\begin{itemize}
\item why is our tool better than the others?
\item random forest approach
\item far more datapoints in training data sets
\item focusing on most important amino acid scales
\end{itemize}

% --------------------------------------------------------------------------- %

\section*{Acknowledgments}

This work was funded by the S\"achsische Aufbaubank (SAB)

\bibliographystyle{unsrt}
\bibliography{ppi_prediction}

\end{document}


